### PR TITLE
[native] Add substitute Windows headers

### PR DESCRIPTION
**Pull request by Joshua-Ashton**
Adds substutite Windows + DirectX headers for use when building DXVK natively for Linux

### CODE REVIEW

**Comment by Joshua-Ashton** [comment]
If there is anything here to review at all really, it's the `winbase.h` I made which just has the substitute Windows defines.

There is also `native_sdl2.h` but that's not very interesting really.

**Comment by Joshua-Ashton** [comment]
For any header license concerns please see `native/directx/readme.txt`.

The relevant part (copied exactly from their COPYING file):

```
DirectX and DDK headers are under GNU LGPLv2.1+ (see the file
COPYING.LGPLv2.1) and copyrighted by various people. Using these
headers doesn't make LGPLv2.1 apply to your code, because these
headers files contain only data structure definitions, short
macros, and short inline functions. Here is the relevant part
from LGPLv2.1 section 5 paragraph 4:

    If such an object file uses only numerical parameters, data
    structure layouts and accessors, and small macros and small
    inline functions (ten lines or less in length), then the use
    of the object file is unrestricted, regardless of whether it
    is legally a derivative work.
```

File: include/native/wsi/native_wsi.h [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#pragma once
```
Comment (doitsujin) [inline include/native/wsi/native_wsi.h:1]: What is this header for and why don't we just include the sdl2 stuff from wsi_platform.h as necessary?

Comment (Joshua-Ashton) [reply include/native/wsi/native_wsi.h:1]: This header is for games/apps built against DXVK native to include.

File: include/native/windows/oaidl.h [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#pragma once
+
+// Don't care.
```
Comment (doitsujin) [inline include/native/windows/oaidl.h:3]: Any reason why we need these empty headers?

File: include/native/windows/windows_base.h [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+// GCC complains about the COM interfaces
+// not having virtual destructors
+
+// and class conversion for C...DESC helper types
+#if defined(__GNUC__) && defined(__cplusplus)
+#pragma GCC diagnostic ignored "-Wnon-virtual-dtor"
+#pragma GCC diagnostic ignored "-Wclass-conversion"
+#endif // __GNUC__ && __cplusplus
+
+struct SDL_Window;
```
Comment (doitsujin) [inline include/native/windows/windows_base.h:21]: do we really need to forward-declare that here of all places?

File: include/native/windows/poppack.h [hunk 4]
```diff
@@ -0,0 +1,8 @@
+/**
+ * This file has no copyright assigned and is placed in the Public Domain.
+ * This file is part of the mingw-w64 runtime package.
+ * No warranty is given; refer to the file DISCLAIMER.PD within this package.
+ */
+#if !(defined(lint) || defined(RC_INVOKED))
+#pragma pack(pop)
+#endif
```
Comment (doitsujin) [inline include/native/windows/poppack.h:8]: does this do anything remotely useful for native builds?

Comment (Joshua-Ashton) [reply include/native/windows/poppack.h:8]: Yes, the struct packing matters.

File: include/native/windows/windows_base.h (hunk 3 above)
Comment (Joshua-Ashton) [reply include/native/windows/windows_base.h:21]: No, this is a left over, will fix.

File: include/native/windows/oaidl.h (hunk 2 above)
Comment (Joshua-Ashton) [reply include/native/windows/oaidl.h:3]: The DX headers include them for common IDL stuff but they don't define anything useful or needed.

**Review by doitsujin — APPROVED** [review]

**Pull request merged**